tests/test_init.py

```diff
@@ -264,6 +264,52 @@ async def test_update_data():
             await tplink.update_data(switch_mac=TEST_SWITCH_MAC)
 
 
+async def test_partial_update_data():
+    """Test update data function with subset."""
+    with patch("tplink_ess_lib.network.socket.socket") as mock_socket:
+        mock_socket = mock_socket.return_value
+        packet1 = bytes.fromhex(base64.b64decode(TEST_PACKETS[9]).decode("utf-8"))
+        packet2 = bytes.fromhex(base64.b64decode(TEST_PACKETS[10]).decode("utf-8"))
+        packet3 = bytes.fromhex(base64.b64decode(TEST_PACKETS[2]).decode("utf-8"))
+        packet5 = bytes.fromhex(base64.b64decode(TEST_PACKETS[4]).decode("utf-8"))
```

Comment:
This code is copied from test_update_data() above.  Maybe the two tests could extract this prep into some constants for reuse....

Reply:
Ya this was a quick cobbling together too

Reply:
I'll fix this in https://github.com/firstof9/tplink-ess-lib/issues/43